### ISSUE TITLE
[Accessibility]: Search results screen reader issues

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#34737]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34190

### Problem description

- [ ] User not notified when results come in or search completes (probably should aria-live on progress element, adding a hidden label to state that the search is in progress or completed)
 ![image](https://user-images.githubusercontent.com/206864/166078867-130bb5ef-d07f-4d8c-8edb-08aac7c85572.png)
- [ ] (i) icon next to result count not labeled and doesn't read the tooltip contents
![image](https://user-images.githubusercontent.com/206864/166078919-12a0c9ed-0c29-4e90-8bf1-2699efbd0a1c.png)
- [ ] When search query is not diff/commit, disabled state "Monitor" button not announced by screen reader
![image](https://user-images.githubusercontent.com/206864/166078959-d3a61cf1-756b-4a86-93cc-38cf9f4e6e50.png)
- [ ] Caret dividing repo from filename should be aria-hidden
![image](https://user-images.githubusercontent.com/206864/166078984-a0598df7-c24c-46c9-bf5b-1dbd47fe07d9.png)
- [ ] Hide/show all matches button read twice by screen reader
![image](https://user-images.githubusercontent.com/206864/166079006-98f7dd75-c3c8-4948-a9dd-0e7873cf373d.png)
- [ ] On type:symbol search results, the type icon next to each symbol is not read by screen reader (should have label based on symbol type)
![image](https://user-images.githubusercontent.com/206864/166079160-10d0e079-a384-4226-a07e-678bf0b2317f.png)
- [ ] On diff/commit search results, git hash and time read together (should have some sort of separation, maybe a hidden comma)
![image](https://user-images.githubusercontent.com/206864/166079191-f4ca2d5c-8415-4a79-b18d-6ce7934f272b.png)
- [ ] Colon between author and commit description should be aria-hidden
![image](https://user-images.githubusercontent.com/206864/166079215-447f727a-98d0-4f8a-a236-8ef0f42e5592.png)
- [ ] Sidebar section collapse button title not read by screen reader (modify aria-label to also have the section title)
![image](https://user-images.githubusercontent.com/206864/166079246-1509d1e1-cf90-4506-a596-ecaf0792dd72.png)
- [ ] Clicking on a link in the sidebar that edits the search query doesn't notify the user that they are now editing the query (should announce this to the user)
- [ ] Links in the sidebar and no results page have an icon that indicated they will open in a new tab, but this icon is not read by the screen reader
![image](https://user-images.githubusercontent.com/206864/166079358-d467a4da-9332-443c-a006-0c649fd5dcc0.png)
- [ ] Video thumbnails on the no results page should be aria-hidden or should be changed to presentation role with empty alt
![image](https://user-images.githubusercontent.com/206864/166079435-b1777eb5-352b-428a-b3e2-1f81a703e94d.png)

### Expected behavior

All the above issues should be fixed

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by GitStart. I have assigned the label 'gitstart' and added the assignee 'gitstart-sourcegraph'.

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.

**Comment by gitstart-sourcegraph** [comment]
Hi @limitedmage 
For this requirement `User not notified when results come in or search completes (probably should aria-live on progress element, adding a hidden label to state that the search is in progress or completed)` want to confirm that we can use `screenReaderAnnounce.ts` util from wildcard.
